worker/transcribee_worker/diarize.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from typing import List, Optional, SupportsFloat, SupportsInt
+
+import torch
+from pyannote.audio import Pipeline
+from pyannote.core.json import dumps
+from transcribee_proto.document import Segment
+
+from .config import settings
+
+
+def ensure_float(x: Optional[SupportsFloat]) -> Optional[float]:
+    if x is None:
+        return None
+    else:
+        return float(x)
+
+
+def ensure_int(x: Optional[SupportsInt]) -> Optional[int]:
+    if x is None:
+        return None
+    else:
+        return int(x)
+
+
+def diarize(audio, progress_callback) -> List[Segment]:
+    pipeline = Pipeline.from_pretrained(
+        "pyannote/speaker-diarization@2.1",
+        use_auth_token=settings.HUGGINGFACE_TOKEN,
+        cache_dir=settings.MODELS_DIR,
+    )
+    device_name = "cpu"
+    if torch.backends.mps.is_available():
```

Comment:
we probably want to do this for other backends as well in the future?

Reply:
Yes, but I can't really test this, so I decided to just use the default. I think we probably want to config option for this in the long run